topic-operator/src/main/java/io/strimzi/operator/topic/v2/KubeRef.java

```diff
@@ -5,18 +5,17 @@
 package io.strimzi.operator.topic.v2;
 
 import io.strimzi.api.kafka.model.KafkaTopic;
-import io.strimzi.operator.common.model.StatusUtils;
 
 import java.util.Objects;
 
 /**
  * Reference to a resource in Kube.
  * Equality is based on {@link #namespace()} and {@link #name()}.
- * {@link #creationTime()} is present to allow disambiguation of multiple KafkaTopics managing the same topic in Kafka.
+ * {@link #creationTimeNs()} is present to allow disambiguation of multiple KafkaTopics managing the same topic in Kafka.
  */
-record KubeRef(String namespace, String name, long creationTime) {
+record KubeRef(String namespace, String name, long creationTimeNs) {
     KubeRef(KafkaTopic kt) {
-        this(kt.getMetadata().getNamespace(), kt.getMetadata().getName(), StatusUtils.isoUtcDatetime(kt.getMetadata().getCreationTimestamp()).toEpochMilli());
+        this(kt.getMetadata().getNamespace(), kt.getMetadata().getName(), System.nanoTime());
```

Comment:
Hi Tom, thanks for looking in to this issue. 

Right, `System.nanoTime` makes the issue less evident, but it doesn't solve it, plus the issues you mentioned.  I also tried to order by `creationTime` and `uid`, but it doesn't work either. The old operator is able to determine the managing topic simply because it reconciles topics one by one, and has the state information of the reconciled topic.

> It shouldn't matter whether A and B get reconciled in the same batch or not

Agree on that, but with regards to the issue. The `creationTime` information that we use to sort refs in `validateSingleManagingResource` is coming from Kubernetes and has low resolution, so there is no guarantee that you will get the oldest ref first (which should be the reconciled ref) when two or more refs have the same `creationTime`. 

In the end, I think we can't fix this problem without keeping persistent state, and we don't want to do that. What we can do, is to change the error message to indicate all resource names with the same `topicName` (including the oldest). This would still help the user to identify the offending topics in case there are thousands of them. Wdyt?